homeassistant/components/fritz/config_flow.py

```diff
@@ -191,16 +200,27 @@ def _show_setup_form_init(
         self, errors: dict[str, str] | None = None
     ) -> ConfigFlowResult:
         """Show the setup form to the user."""
+
+        if self.show_advanced_options:
+            advanced_data_schema = {
+                vol.Optional(CONF_PORT): vol.Coerce(int),
+            }
+        else:
+            advanced_data_schema = {}
+
+        data_schema = vol.Schema(
+            {
+                vol.Optional(CONF_HOST, default=DEFAULT_HOST): str,
+                **advanced_data_schema,
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Optional(CONF_SSL, default=DEFAULT_SSL): bool,
+            }
+        )
```

Comment:
no need to define `data_schema` , just use the `vol.Schema` in the `async_show_form` as it was before